packages/synapse-react-client/src/lib/containers/DialogBase.tsx

```diff
@@ -0,0 +1,82 @@
+import CloseIcon from '@mui/icons-material/Close'
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogProps,
+  DialogTitle,
+  IconButton,
+  Stack,
+  SxProps,
+} from '@mui/material'
+import React from 'react'
+import { HelpPopover } from './HelpPopover'
+
+export type CloseButtonProps = {
+  sx?: SxProps
+  onClick?: () => void
+}
+
+export const CloseButton: React.FC<CloseButtonProps> = ({
+  sx = { color: 'grey.700' },
+  onClick,
+}) => {
+  return (
+    <IconButton sx={sx} onClick={onClick} aria-label="close">
+      <CloseIcon />
+    </IconButton>
+  )
+}
+
+export type DialogBaseProps = {
+  open: boolean
+  title: string | JSX.Element
+  content: string | JSX.Element
+  actions?: JSX.Element
+  className?: string
+  onCancel: () => void
+  hasCloseButton?: boolean
+  helpMarkdown?: string
+  helpUrl?: string
+  maxWidth?: DialogProps['maxWidth']
```

Comment:
We will probably also want to expose `fullWidth` for the EntityFinderModal but that change can wait

Reply:
Easy enough to add now, will be in the next commit!